### PR TITLE
Use ruff format

### PR DESCRIPTION
**Pull request by karpetrosyan**
Refs: https://github.com/encode/httpx/pull/2901

### CODE REVIEW

File: scripts/check
Comment (karpetrosyan) [inline scripts/check:13]:
```suggestion
${PREFIX}ruff format $SOURCE_FILES_WITHOUT_SYNC --diff
```

File: pyproject.toml
Comment (T-256) [inline pyproject.toml:109]: it is the default; you can remove it.

File: pyproject.toml
Comment (T-256) [inline pyproject.toml:117]: I suggest use [tool.ruff.pycodestyle.max-line-length](https://docs.astral.sh/ruff/settings/#pycodestyle-max-line-length) with value of 120 instead of ignore it.

File: requirements.txt
Comment (T-256) [inline requirements.txt:17]:
```suggestion
ruff==0.1.3
```

File: scripts/check
Comment (tomchristie) [inline scripts/check:13]: The `SOURCE_FILES_WITHOUT_SYNC=...` is a bit opaque.
Can we use a less complex approach?

Comment (karpetrosyan) [reply scripts/check:13]: I believe we can entirely remove it.
Ruff will use the "excludes" setting from the `pyproject.toml` file.

Demo:

``` shell
ruff httpcore/ tests --verbose

[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Ignored path via `exclude`: "/home/test/programs/gitprojects/httpcore/tests/_sync"
[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Included path via `include`: "/home/test/programs/gitprojects/httpcore/httpcore/_synchronization.py"
[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Included path via `include`: "/home/test/programs/gitprojects/httpcore/httpcore/_exceptions.py"
[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Ignored path via `exclude`: "/home/test/programs/gitprojects/httpcore/httpcore/_sync"
[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Included path via `include`: "/home/test/programs/gitprojects/httpcore/httpcore/_utils.py"
[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Included path via `include`: "/home/test/programs/gitprojects/httpcore/tests/test_models.py"
[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Included path via `include`: "/home/test/programs/gitprojects/httpcore/httpcore/_api.py"
[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Included path via `include`: "/home/test/programs/gitprojects/httpcore/httpcore/_backends/__init__.py"
[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Included path via `include`: "/home/test/programs/gitprojects/httpcore/httpcore/_backends/trio.py"
[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Included path via `include`: "/home/test/programs/gitprojects/httpcore/tests/test_api.py"
[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Included path via `include`: "/home/test/programs/gitprojects/httpcore/httpcore/__init__.py"
```

**Review by T-256 — APPROVED** [review]

File: pyproject.toml
Comment (karpetrosyan) [reply pyproject.toml:118]: Sure, thanks

**Comment by karpetrosyan** [comment]
Now adding the `max-line-length` option does not make any sense because `httpx` was switched to the default ruff configuration in https://github.com/encode/httpx/pull/2922. 

There are some files that should be reformated with the new configurations, but to keep the PR simple and **reviewable**, I will update these lines after any approvals from the @encode/maintainers.

**Comment by tomchristie** [comment]
Okay, so the linting checks are currently failing here.

I'd suggest we *keep* the max-line-length and get this merge on that basis, and then have a separate PR removing that and dealing with the line length changes that are required as a result.

**Review by T-256 — APPROVED** [review]

**Review by tomchristie — APPROVED** [review]

**Pull request merged**
